### ISSUE TITLE
Positional parameters cannot be used to specify probes unless they are already a part of a string

### ISSUE THREAD

**Issue by 0mp** [iovisor/bpftrace#2513]
### What reproduces the bug?

The following script works as expected:

```console
$ sudo bpftrace -c "/bin/cat /etc/shells" -e 'k:$1 {printf("%s\n", probe);}' fput
```

However, if we decide to specify the whole probe with a positional argument, the script fails:

```console
$ sudo bpftrace -c "/bin/cat /etc/shells" -e '$1 {printf("%s\n", probe);}' k:fput
FATAL: Invalid probe type made it to attachpoint parser
Aborted
```

### bpftrace --info

```
System
  OS: Linux 5.4.0-139-generic #156-Ubuntu SMP Fri Jan 20 17:27:18 UTC 2023
  Arch: x86_64

Build
  version: v0.17.0
  LLVM: 12.0.0
  unsafe uprobe: no
  bfd: yes
  libdw (DWARF support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: no
  probe_read_user_str: no
  probe_read_kernel: no
  probe_read_kernel_str: no
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes
  get_boot_ns: no
  dpath: no
  skboutput: no

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: no
  map batch: no
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: no
  iter:task: no
  iter:task_file: no
  kprobe_multi: no
  raw_tp_special: no
```

**Comment by viktormalik** [comment]
Hi, thanks for reporting this! Should be fixed by #2514, feel free to give it a try.

**Comment by 0mp** [comment]
Thanks, @viktormalik, I'll take a look!
